core/domain/stats_domain.py

```diff
@@ -403,6 +405,155 @@ def validate(self):
+        """Constructs an ExplorationIssue domain object.
+
+        Args:
+            issue_id: str. ID of the Issue.
+            schema_version: int. Schema version of the customization args dict.
+            customization_args: dict. The customization dict. The keys are
+                names of customization_args and the values are dicts with a
+                single key, 'value', whose corresponding value is the value of
+                the customization arg.
+        """
+        self.issue_id = issue_id
+        self.schema_version = schema_version
+        self.customization_args = customization_args
+
+    def to_dict(self):
+        """Returns a dict representation of the ExplorationIssue domain object.
+
+        Returns:
+            dict. A dict mapping of all fields of ExplorationIssue object.
+        """
+        return {
+            'issue_id': self.issue_id,
+            'schema_version': self.schema_version,
+            'customization_args': (
+                {} if self.issue_id is None else
+                exp_domain.get_full_customization_args(
+                    self.customization_args,
+                    issue_registry.Registry.get_issue_by_id(
+                        self.issue_id).customization_arg_specs))
+        }
+
+    @classmethod
+    def from_dict(cls, issue_dict):
+        """Returns an ExplorationIssue object from a dict.
+
+        Args:
+            issue_dict: dict. A dict mapping of all fields of ExplorationIssue
+                object.
+
+        Returns:
+            ExplorationIssue. The corresponding ExplorationIssue domain object.
+        """
+        return cls(
+            issue_dict['issue_id'],
+            issue_dict['schema_version'],
+            issue_dict['customization_args'])
+
+    def validate(self):
+        """Validates the ExplorationIssue domain object."""
+        if not isinstance(self.issue_id, basestring):
+            raise utils.ValidationError(
+                'Expected issue_id to be a string, received %s' % (
+                    self.issue_id))
+
+        if not isinstance(self.schema_version, int):
+            raise utils.ValidationError(
+                'Expected schema_version to be an int, received %s' % (
+                    self.schema_version))
+
+        try:
+            issue = issue_registry.Registry.get_issue_by_id(
+                self.issue_id)
+        except KeyError:
+            raise utils.ValidationError('Invalid issue ID: %s' % self.issue_id)
+
+        exp_domain.validate_customization_args_and_values(
+            'issue', self.issue_id, self.customization_args,
+            issue.customization_arg_specs)
+
+
+class LearnerAction(object):
+    """Domain object representing a learner action."""
+
+    def __init__(self, action_id, schema_version, customization_args):
+        """Constructs a LearnerAction domain object.
+
+        Args:
+            action_id: str. ID of the Learner Action.
```

Comment:
nit: change to `ID of the action` to fit in w/ the other domain objects.

Reply:
Done.